sys/userns_linux.go

```diff
@@ -0,0 +1,39 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package sys
+
+import (
+	"github.com/opencontainers/runc/libcontainer/user"
+)
+
+// RunningInUserNS detects whether we are currently running in a user namespace.
+// Originally copied from github.com/lxc/lxd/shared/util.go
+func RunningInUserNS() bool {
+	uidmap, err := user.CurrentProcessUIDMap()
```

Comment:
Given that we only use this for detection, but not using it otherwise; wondering if we should simplify `RunningInUserNS` to go back the original implementation in https://github.com/lxc/lxd/blob/8447da505e42763c1fb56e2ab905aa8146b29030/shared/util.go#L690-L710

@AkihiroSuda wdyt?